File: backend/bookstore.API/Models/Book.cs
namespace bookstore.API.Models;

// Use this model to represent each book coming from the database.
public class Book
{
    public int BookID { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Author { get; set; } = string.Empty;

    public string Publisher { get; set; } = string.Empty;

    public string ISBN { get; set; } = string.Empty;

    public string Classification { get; set; } = string.Empty;

    public string Category { get; set; } = string.Empty;

    public int PageCount { get; set; }

    public double Price { get; set; }
}

File: frontend/src/types/book.ts
// Keep the frontend aligned with the shape of each book from the API.
export interface Book {
  bookID: number;
  title: string;
  author: string;
  publisher: string;
  isbn: string;
  classification: string;
  category: string;
  pageCount: number;
  price: number;
}

// Keep the frontend aligned with the paginated response from the books endpoint.
export interface BookResponse {
  books: Book[];
  totalNumBooks: number;
}
